grafana-operator/grafana-operator
# fix: use non-fqdn url for grafana to support non-standard cluster names

## Description

Currently, grafana-operator uses FQDN names for its calls to grafana API (e.g. `http://grafana.monitoring.svc.cluster.local:3000`), which is incompatible with custom cluster names.
For internal DNS requests, there's no need to rely on FQDN, `<svc-name>.<namespace>:<port>` format is perfectly fine.
The PR addresses that.

Additional changes:
- Some code style issues are automatically fixed by `gofumpt`;
- Fixed staticcheck complaint about error formatting.

## Relevant issues/tickets

Fixes: https://github.com/grafana-operator/grafana-operator/issues/592

## Type of change


## Checklist


## Verification steps

1. Deploy grafana instance with any spec
2. Make sure operator is able to communicate with grafana:
- For clusters with standard names, operator should work just as before;
- For clusters with a custom name, operator should start working.
